File: board_dsl/src/main/java/com/dooho/board/api/auth/controller/SearchController.java
package com.dooho.board.api.auth.controller;

import com.dooho.board.api.board.search.SearchDto;
import com.dooho.board.api.ResponseDto;
import com.dooho.board.api.board.BoardEntity;
import com.dooho.board.api.board.search.SearchEntity;
import com.dooho.board.api.board.search.SearchService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/search")
public class SearchController {

    private final SearchService searchService;

    @Autowired
    public SearchController(SearchService searchService) {
        this.searchService = searchService;
    }


    @PostMapping("")
    public ResponseDto<List<BoardEntity>> searchBoard(@RequestBody SearchDto requestBody){
        ResponseDto<List<BoardEntity>> result = searchService.getSearchList(requestBody);
        return result;
    }

    @GetMapping("/popularSearchList")
    public ResponseDto<List<SearchEntity>> getPopularSearchList(){
        ResponseDto<List<SearchEntity>> result = searchService.getPopularSearchList();
        return result;
    }
}

File: board_spring/src/main/java/com/dooho/board/api/board/search/SearchController.kt
package com.dooho.board.api.board.search

import com.dooho.board.api.ResponseDto
import com.dooho.board.api.board.BoardEntity
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/search")
class SearchController @Autowired constructor(private val searchService: SearchService) {
    @PostMapping("")
    fun searchBoard(@RequestBody requestBody: SearchDto?): ResponseDto<List<BoardEntity?>?> {
        return searchService.getSearchList(requestBody)
    }

    @get:GetMapping("/popularSearchList")
    val popularSearchList: ResponseDto<List<SearchEntity?>?>
        get() = searchService.popularSearchList
}